pkg/trace/api/otlp.go

```diff
@@ -398,6 +398,54 @@ func marshalEvents(events ptrace.SpanEventSlice) string {
 	return str.String()
 }
 
+// marshalLinks marshals span links into JSON.
+func marshalLinks(links ptrace.SpanLinkSlice) string {
+	var str strings.Builder
+	str.WriteString("[")
+	for i := 0; i < links.Len(); i++ {
+		l := links.At(i)
+		if i > 0 {
+			str.WriteString(",")
+		}
+		t := l.TraceID()
+		str.WriteString(`{"trace_id":"`)
+		str.WriteString(hex.EncodeToString(t[:]))
+		s := l.SpanID()
+		str.WriteString(`","span_id":"`)
+		str.WriteString(hex.EncodeToString(s[:]))
+		str.WriteString(`"`)
+		if ts := l.TraceState().AsRaw(); len(ts) > 0 {
+			str.WriteString(`,"trace_state":"`)
+			str.WriteString(ts)
+			str.WriteString(`"`)
+		}
+		if l.Attributes().Len() > 0 {
+			str.WriteString(`,"attributes":{`)
+			b := false
+			l.Attributes().Range(func(k string, v pcommon.Value) bool {
+				if b {
+					str.WriteString(",")
+				}
+				b = true
+				str.WriteString(`"`)
+				str.WriteString(k)
+				str.WriteString(`":"`)
+				str.WriteString(v.AsString())
+				str.WriteString(`"`)
+				return true
+			})
+			str.WriteString("}")
+		}
+		if l.DroppedAttributesCount() > 0 {
```

Comment:
I would add a dedicated suite of unit tests specifically for this function to ensure all possible variations are correctly converted and all commas are placed correctly, e.g.:
* With one link
* With multiple links
* With one attribute
* With multiple attributes
* Without attributes
* With dropped attribute count
* Without dropped attribute count

And combinations of the above.

Reply:
Done.